graylog2-server/src/main/java/org/graylog2/indexer/IndexMapping.java

```diff
@@ -104,8 +111,13 @@ protected Map<String, Map<String, Object>> fieldProperties(String analyzer) {
                 .put(Message.FIELD_GL2_MESSAGE_ID, notAnalyzedString())
                 // to support wildcard searches in source we need to lowercase the content (wildcard search lowercases search term)
                 .put("source", analyzedString("analyzer_keyword", true))
-                .put("streams", notAnalyzedString())
-                .build();
+                .put("streams", notAnalyzedString());
+
+        if (customFieldMappings != null) {
+            customFieldMappings.forEach(customMapping -> builder.put(customMapping.fieldName(), type(customMapping.physicalType())));
```

Comment:
I think we should use our logical types when configuring the mapping and map them to their physical counterparts in here when creating the mapping template.